Mathlib/Analysis/InnerProductSpace/Laplace.lean

```diff
@@ -0,0 +1,180 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.Analysis.Calculus.ContDiff.Basic
+import Mathlib.Analysis.Calculus.ContDiff.Operations
+import Mathlib.Analysis.InnerProductSpace.CanonicalTensor
+
+/-!
+# The Laplacian
+
+This file defines the Laplacian for functions `f : E → F` on real, finite-dimensional, inner product
+spaces `E`. In essence, we define the Laplacian of `f` as the second derivative, applied to the
+canonical covariant tensor of `E`, as defined and discussed in
+`Mathlib.Analysis.InnerProductSpace.CanonicalTensor`.
+
+We show that the Laplacian is `ℂ`-linear on continuously differentiable functions, and establish the
+standard formula for computing the Laplacian in terms of orthonormal bases of `E`.
+-/
+
+open Filter InnerProductSpace TensorProduct Topology
+
+section secondDerivativeAPI
+
+/-!
+## Supporting API
+
+The definition of the Laplacian of a function `f : E → F` involves the notion of the second
+derivative, which can be seen as a continous multilinear map `ContinuousMultilinearMap 𝕜 (fun (i :
+Fin 2) ↦ E) F`, a bilinear map `E →ₗ[𝕜] E →ₗ[𝕜] F`, or a linear map on tensors `E ⊗[𝕜] E →ₗ[𝕜]
+F`. This section provides convenience API to convert between these notions.
+-/
+
+variable
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  {F : Type*} [NormedAddCommGroup F] [NormedSpace 𝕜 F]
+  {G : Type*} [NormedAddCommGroup G] [NormedSpace 𝕜 G]
+
+variable (𝕜) in
+/--
+Convenience reformulation of the second iterated derivative, as a map from `E` to bilinear maps
+`E →ₗ[ℝ] E →ₗ[ℝ] ℝ
+-/
+noncomputable def bilinear_of_iteratedFDeriv_two (f : E → F) : E → E →ₗ[𝕜] E →ₗ[𝕜] F :=
+  fun x ↦ (fderiv 𝕜 (fderiv 𝕜 f) x).toLinearMap₂
+
+/--
+Expression of `bilinear_of_iteratedFDeriv_two` in terms of `iteratedFDeriv`.
+-/
+lemma bilinear_of_iteratedFDeriv_two_eq_iteratedFDeriv (f : E → F) (e e₁ e₂ : E) :
+    bilinear_of_iteratedFDeriv_two 𝕜 f e e₁ e₂ = iteratedFDeriv 𝕜 2 f e ![e₁, e₂] := by
+  simp [iteratedFDeriv_two_apply f e ![e₁, e₂], bilinear_of_iteratedFDeriv_two]
+
+variable (𝕜) in
+/--
+Convenience reformulation of the second iterated derivative, as a map from `E` to linear maps
+`E ⊗[𝕜] E →ₗ[𝕜] F`.
+-/
+noncomputable def tensor_of_iteratedFDeriv_two (f : E → F) : E → E ⊗[𝕜] E →ₗ[𝕜] F :=
+  fun e ↦ lift (bilinear_of_iteratedFDeriv_two 𝕜 f e)
+
+/--
+Expression of `tensor_of_iteratedFDeriv_two` in terms of `iteratedFDeriv`.
+-/
+lemma tensor_of_iteratedFDeriv_two_eq_iteratedFDeriv (f : E → F) (e e₁ e₂ : E) :
+    tensor_of_iteratedFDeriv_two 𝕜 f e (e₁ ⊗ₜ[𝕜] e₂) = iteratedFDeriv 𝕜 2 f e ![e₁, e₂] := by
+  rw [← bilinear_of_iteratedFDeriv_two_eq_iteratedFDeriv, tensor_of_iteratedFDeriv_two]
+  rfl
+
+end secondDerivativeAPI
+
+/-!
+## Definition of the Laplacian
+
+Use `open InnerProductSpace` to access the notation `Δ` for `InnerProductSpace.Laplacian`.
```

Comment:
Can you move this into the doc-string of `Laplacian` itself? This way, if a user hovers over the notation (or the declaration), they also see this mention of the notation.